spark/src/main/scala/org/apache/comet/rules/RewriteJoin.scala

```diff
@@ -65,9 +65,8 @@ object RewriteJoin extends JoinSelectionHelper {
   def rewrite(plan: SparkPlan): SparkPlan = plan match {
     case smj: SortMergeJoinExec =>
       getSmjBuildSide(smj) match {
-        case Some(BuildRight) if smj.joinType == LeftSemi =>
-          // TODO this was added as a workaround for TPC-DS q14 hanging and needs
-          // further investigation
```

Comment:
I no longer see the issue with LeftSemi + BuildRight, so this fallback is removed